File: Client/files/blip.js
import * as alt from 'alt-client';
import * as native from 'natives';
var totalBlips = 0;
var blips = new Array();
alt.onServer('createLocalBlip', (blipJson) => {
    var blipInfo = JSON.parse(blipJson);
    alt.log("Creating Blip: " + blipInfo.Name);
    let blip = native.addBlipForCoord(blipInfo.PosX, blipInfo.PosY, blipInfo.PosZ);
    native.setBlipSprite(blip, blipInfo.Sprite);
    native.setBlipColour(blip, blipInfo.Color);
    native.setBlipScale(blip, blipInfo.Scale);
    native.setBlipAsShortRange(blip, blipInfo.ShortRange);
    native.beginTextCommandSetBlipName('STRING');
    native.addTextComponentSubstringPlayerName(blipInfo.Name);
    native.endTextCommandSetBlipName(blip);
    if (blipInfo.UniqueId == -1) {
        totalBlips += 1;
        blipInfo.UniqueId = `${totalBlips}`;
    }
    if (blips[blipInfo.UniqueId] !== undefined) {
        native.removeBlip(blips[blipInfo.UniqueId]);
    }
    blips[blipInfo.UniqueId] = blip;
});
alt.onServer('deleteLocalBlip', (uniqueId) => {
    if (blips[uniqueId] !== undefined) {
        native.removeBlip(blips[uniqueId]);
        delete blips[uniqueId];
    }
});
alt.onServer('deleteAllBlips', () => {
    blips.forEach((blip) => {
        native.removeBlip(blip);
    });
    blips = new Array();
});

File: Server/Objects/Blip.cs
using AltV.Net.Data;
using AltV.Net.Elements.Entities;
using Newtonsoft.Json;

namespace Server.Objects
{
    public class Blip
    {
        public float PosX { get; set; }
        public float PosY { get; set; }
        public float PosZ { get; set; }

        public int Sprite { get; set; }

        public int Color { get; set; }

        public float Scale { get; set; }

        public bool ShortRange { get; set; }

        public string Name { get; set; }

        public int? UniqueId { get; set; }

        /// <summary>
        /// Creates a new blip
        /// </summary>
        /// <param name="name"></param>
        /// <param name="position"></param>
        /// <param name="sprite"></param>
        /// <param name="color"></param>
        /// <param name="scale"></param>
        /// <param name="shortRange"></param>
        /// <param name="uniqueId"></param>
        public Blip(string name, Position position, int sprite, int color, float scale, bool shortRange = true,
            int uniqueId = -1)
        {
            Name = name;
            PosX = position.X;
            PosY = position.Y;
            PosZ = position.Z;
            Sprite = sprite;
            Color = color;
            Scale = scale;
            ShortRange = shortRange;
            UniqueId = uniqueId;
        }
    }
}